### PR TITLE
New chem RadAway: a slightly toxic radiation resisting medicine that can scrub contamination when sprayed onto people or objects

### PR DESCRIPTION
**Pull request by Chubbygummibear**
I HATE RADIATION

I HATE RADIATION

https://user-images.githubusercontent.com/46236974/162529008-6c65459c-a574-410c-a885-b89459b73fd3.mp4

This video is a bit of an exaggeration, as the values have been nerfed so that items don't instantly contaminate, only people do

# Document the changes in your pull request

Adds a new reagent named RadAway which is a slightly toxic radiation scrubbing solution. Ingesting is is a bad idea as it will causes slight toxin damage over time. However,  the REAL strength of this chem is that when applied via touch or spray, it cleanses radioactive contamination. So if you've ever seen medbay looking like this before...
image
This chem would actually allow you to combat the deathly green glow.
Items and objects sprayed down by RadAway will have a permanently increased radiation insulation as well to prevent immediate recontamination, however the insulation buff on people only lasts as long as it's in your system.

# Wiki Documentation

Wake up wiki team, new chem dropped

Radaway deals 1 toxin per tick, buffs your radiation insulation while processing, and makes you vomit when it leaves your system (but the vomit doesn't stun)

cleans radioactive contamination from people and slowly from objects when applied via touch or spray and buffs items' rad insulation if it's not already protected

Recipe:
1 part potassium iodide
1 part seiver
1 part space cleaner
Chill to 200k

Results in 3u

# Changelog

:cl:  
rscadd: New radiation treatment chem: RadAway. A radioactive contamination scrubbing chem that can clean contamination (the green glow) off of items and people that are sprayed or splashed with it
tweak: The lavaland ruin that had a bottle of potassium iodide renamed rad-away gets a bottle of this RadAway instead
/:cl:

### CODE REVIEW

**Comment by stwordcitizen** [comment]
Only thing i'd suggest is that it shouldn't instantly remove contamination, maybe make it take some time before it instantly removes the objects rads.

**Comment by tattax** [comment]
i suggest the above + only working on structures + harder to make + no insta stun when applied what the hell

**Comment by JamieD1** [comment]
Bit op innit?

**Comment by Chubbygummibear** [comment]
> Only thing i'd suggest is that it shouldn't instantly remove contamination, maybe make it take some time before it instantly removes the objects rads.

I could maybe make this work on a timer for after spraying, because as it stands items don't actually have radiation, the get contaminated which is a component that stores the strength of the radiation that impacted them to the point and rolled unfortunately enough to have the contamination component added. so the best i can think of is just a delay, but i'm not sure if reducing the power until it's gone is possible.


> i suggest the above + only working on structures + harder to make + no insta stun when applied what the hell

structures are actually pretty incredibly safe from getting contaminated due to having a higher natural rad insulation which is why i don't even have a structure scrubbing proc on the reagent. harder to make is a valid point, but i wanted it to be somewhat simple enough for less skilled chemists to make in a pinch, but not so complex like pentetic acid because if medbay is already green, unless it's a very skilled cmo everyone will probably already be rad cooked. and the insta stun isn't an effect of the chem, that's just the radiation poisoning procing at about the same time as the application lmao


> Bit op innit?

maybe, but i'm determined to add something that scrubs contamination because currently the ONLY thing in the game that does this is the shower and that does it too slowly to matter, and you can't shove the telecaster or lightswitch in the shower. i've seen Chernobyl-bay a few times now and i hate not having toolsl to deal with it outside of "heal faster than it cooks you"

**Comment by LazennG** [comment]
seen a lot of rounds fucked over by radiation lately with 0 efforts by the crew to fix it with rad collectors, so I think this is a good thing to have so rads aren't a permanent issue if engineers and cargo are unhelpful.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -1757,6 +1757,43 @@
...
+	..()
+	//store the person's original insulation so they're only extra protected while it's in their system
+	old_insulation = L.rad_insulation
+	L.rad_insulation = RAD_LIGHT_INSULATION
+
+/datum/reagent/medicine/radaway/on_mob_end_metabolize(mob/living/L)
+	L.rad_insulation = old_insulation
+	if(iscarbon(L))
+		var/mob/living/carbon/C = L
+		C.vomit() //it binds with the radioactive particles inside you, and they have to come out somehow
```
Comment (SomeguyManperson) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: no
no no no
no no no no no
no
you don't want to see me use this
you don't want to see ANYONE use this
either vomit(stun = FALSE) or do something else

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 2]
```diff
@@ -1757,6 +1757,43 @@
...
+	taste_description = "metallic dust"
+	self_consuming = TRUE
+
+/datum/reagent/medicine/radaway/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(method == TOUCH || method == VAPOR)
+		M.wash(CLEAN_RAD) //you only get decontaminated if it's spray based, can't spam out 100 1u pills
+	
+/datum/reagent/medicine/radaway/on_mob_life(mob/living/carbon/M)
+	if(M.radiation > 0)
+		M.radiation -= min(M.radiation, 15)
```
Comment (SomeguyManperson) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: this isn't contamination and we have other radiation removing chemicals like seiver

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (Chubbygummibear) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: added the stun = false clause

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 2 above)
Comment (Chubbygummibear) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: took out the rad removal part

**Comment by Chubbygummibear** [comment]
~~hmm maybe i should move this to a different reagent category like with space cleaner if it doesn't have any innate healing effects and is effectively a radiation cleaning chem~~

nevermind i'm dumb it gives the rad insulation on metabolization, that's still kinda mediciney

**Comment by tattax** [comment]
>that's just the radiation poisoning procing at about the same time as the application lmao

i meant the vomit part, also by structures i meant machinery and stuff, not objects nor humans since decontaminating those things are indeed close to impossible

**Comment by ToasterBiome** [comment]
I like the idea, and it can always be tweaked later if it is _too easy_ to make.

**Comment by Hopekz** [comment]
> Bit op innit?

yeah using radiation as a weapon is OP

**Comment by ToasterBiome** [comment]
I also think this should remove some radiation, not all of it btw

**Comment by Moja10** [comment]
Removing all is way too strong, that's not how radiation should work. It should scale with how badly something is irradiated. Weak rads need one spray, heavy rads should need 5+ or something. Could make it percentage based, or percentage based + fixed rad removal number so that it can 100% clean.

**Comment by Moja10** [comment]
Also, the recipe is way too weak. 3 clicks from dispenser + chill = enough spray to clean two whole departments

**Comment by Chubbygummibear** [comment]
ok, new update makes it so that when you spray items/objects/machinery with radaway, it saps the strength of the contamination, and applies a shower level scrub. weak contamination will be immediately deleted but stronger will require multiple sprays or the smoke machine to douse it. in addition, i changed the water portion of the recipe to be space cleaner since it's a cleaning agent kinda deal, so that adds another click to the process. I still don't really know how to equate power to clicks in a meaningful way, so +1 click for now

**Comment by MenacingManatee** [comment]
I like the idea, but it seems a little easy to make. The swap from water to space cleaner helps a lot since that massively increases the complexity, and I like that it needs to be chilled instead of heated, since that's uncommon. I'd suggest swapping the chlorine for vodka, or at least adding vodka to the recipe in addition to the chlorine. Vodka is easy enough to get and does have anti-rad properties, but takes enough effort to get it adds some complexity to it.

**Comment by MenacingManatee** [comment]
Pentetic acid in place of potassium iodide would also increase the complexity quite a bit more if you wanted this to be done all in chemistry

**Comment by Chubbygummibear** [comment]
> Pentetic acid in place of potassium iodide would also increase the complexity quite a bit more if you wanted this to be done all in chemistry

so I definitely looked at pentetic, but i feel like it would require a seasoned chemist to make in a meaningful amount of time. and since this is mostly a reactionary chem to a radiation outbreak, I fear that medbay would be entirely cooked by the time someone who's less skilled could make it. What about having seiver as the last component? it's not too hard to make, and the chilled aspect coincides thematically with cold seiver killing radiation?

**Comment by Chubbygummibear** [comment]
tweaked the recipe to be
1u potassium iodide
1u seiver
1u space cleaner
chill to 200k
gets you 3 units of radaway

requires a bit more clicking and knowledge, but i want it to be doable from medbay for when the chairs start glowing

**Comment by JamieD1** [comment]
How about we make this toxic to being sprayed on people so they cant use it on themselves to literally be gods and ignore rads.

**Comment by Chubbygummibear** [comment]
> How about we make this toxic to being sprayed on people so they cant use it on themselves to literally be gods and ignore rads.


but it is toxic? it deals toxin damage per tick for being in your system. and you only get the rad_insulation while it's in your system dealing damage to you. the reason i moved it out of the toxins category was because i was unhappy with how quickly it killed your liver for being in your system, and gas application to an area was lethal to livers

additionally, it doesn't let you ignore rads, it is a tool to remove and prevent CONTAMINATION, the green glow. if you're glowing green with 5k rads that are poisoning and killing you and you get sprayed, your glow will stop but you'll still have 5k rads poisoning you. additionally you have to be thorough in that items in inventories aren't immediately disinfected unless dumped out and sprayed. not even a gas grenade will remove all the contamination due to layering inventories. 

Stworld brought up the point about it being too easy to neuter weaponized radiation since a single spray on a person turns off the contamination pulses, but it wouldn't decontaminate your clothes immediately, or items in your backpack, or inventories in your backpack, like boxes, and if you have a modular tablet or item with smaller parts in it like a BRPED would, all of those would be contaminated without a quick way to disinfect, so i don't believe it's a 1-click counter, even as a gas bomb or smoke machine.

The reason i believe this is necessary for the game is because when departments turn glowing green. all the items and objects constantly bounce rad pulses off one another, perpetuating the contamination, and the shower is the only method that cleanses this, but it works slowly, and doesn't allow you to put the firelocks or tables in the shower to stop the area from slow roasting everyone, and you're stuck there for the time being. Even after getting doused with radaway, you still need to treat the radiation sickness with seiver, pentetic acid, or a LOT of potassium iodide

TL;DR it doesn't treat radiation, it treats contamination, which emits radiation constantly. you can't ignore rads, it makes you resistant to contamination while it's in your system. you take toxin damage while it's in your system. it only 1 click cleanses people, but not the potential radiation supernova emitters in their pockets/bag. even if you're not glowing green, you can still have 5k rads and super cancer and requires more treatment

**Comment by ToasterBiome** [comment]
The points that Chubby brings up are solid. I have no reason to believe that it will be ridiculously overpowered. As they described in their explanation, this won't immediately invalidate weaponized radiation because you're still sick from it, your items are still covered in it, and you're constantly taking damage if you're insulated against the green medbay attack. To me, this feels like the natural progression of combatting radiation. This is not the be all end all of rad fighting, but just one of many in a doctor's toolkit which is sorely needed from the rounds that @stwordcitizen has been uh... active in. I think the crafting recipe is unique and complex enough that you aren't pumping out 500u of it to spray everything, and as Chubby described with the layering inventory, doesn't seem like it's even the best option in most situations. You are still going to kill a lot of people. Medbay will still be ridiculously green. Slimes will still be bright green AFKing in vents under the sleeper. This just helps clearing the invisible killer (all the other ways to get rid of it suck anyway). I think it at least warrants a test merge so we can see in action how players react to it.

**Comment by SomeguyManperson** [comment]
Just due to how easy it is to stack massive amounts of rads having a similarly approachable way to mass remove them is necessary
Unless we want to massively rework radiation in some way or commit a tg having a way to easily combat mass contamination is our best bet

**Review by ghost — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]
